File: OverlayManager/TS/DragonH/Attack.ts
class Attack {
  description: string;
  reach: number;
  range: number;
  rangeMax: number;
  plusToHit: number = 0;
  numTargets: number = 1;
  mods: Array<Mod> = Array<Mod>();
  damages: Array<Damage> = Array<Damage>();
  savingThrow: SavingThrow;
  successfulSaveDamages: Array<Damage> = Array<Damage>();
  constructor(public name: string) {

  }

  addDamage(damageType: DamageType, damageRoll: string): any {
    let damage: Damage = new Damage(damageType, damageRoll);
    this.damages.push(damage);
    return damage;
  }
}

File: DHDM/DnD Model/Attack.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace DHDM
{
	public class Attack
	{
		public string description;
		public int reach;
		public int range;
		public int rangeMax;
		public int plusToHit;
		public int numTargets = 1;
		public List<Mod> mods = new List<Mod>();
		public List<Damage> damages = new List<Damage>();
		public List<Damage> successfulSaveDamages = new List<Damage>();
		public SavingThrow savingThrow;
		Attack(string name)
		{
			Name = name;
		}

		public Damage AddDamage(DamageType damageType, string damageRoll)
		{
			Damage damage = new Damage(damageType, damageRoll);
			damages.Add(damage);
			return damage;
		}

		public string Name { get; set; }
	}
}
